### ISSUE TITLE
Webapp crashes

### ISSUE THREAD

**Issue by johnrubinstein** [cryoem-uoft/cryosparc-issues#93]
@alipunjani @suhaildawood Quite a few times (~5 in the past week or so) I've found the webapp is down. Logging in to the computer running cryosparc I find the webapp is exited and meteor is stopped. Stopping/starting cryoSPARC gets everything back up and running. Is there an error log that would give more info. The following  is from around 6pm today (I don't think it has been used since Monday morning):
**********

cryosparc@turing:~$ cryosparc status
Running CryoSPARC Tool at /home/cryosparc/cryosparc
Current cryoSPARC version: v0.2.31

Process status:

meteor                           STOPPED   Not started
mongod                           RUNNING   pid 15602, uptime 1 day, 20:09:24
webapp                           EXITED    Dec 13 10:53 AM

Config variables:

export CRYOSPARC_ROOT_DIR="/home/cryosparc/cryosparc"
export CRYOSPARC_BASE_PORT=38000
export CRYOSPARC_MONGO_PORT=38001
export CRYOSPARC_MASTER_HOSTNAME="turing.sickkids.ca "
export CRYOSPARC_INSTALL_TYPE=master
export CRYOSPARC_DEVELOP=false
export CRYOSPARC_SUPERVISOR_SOCK_FILE=/tmp/supervisor-3522675935842350195.sock

export CRYOSPARC_CACHE_CUSHION=1024.0
export CRYOSPARC_RAM_SLOTS="4"
export CRYOSPARC_CUDA_DEVS="0"
export CRYOSPARC_HTTP_PORT=38000

cryosparc@turing:~$ cryosparc stop
Stopping cryoSPARC..
mongod: stopped
webapp: ERROR (not running)
Shut down
cryosparc@turing:$ cryosparc start
Starting cryoSPARC..
mongod: started
webapp: started
-----------------------------------------------------

CryoSPARC started. 
 From this machine, access at
    http://localhost:38000
 From other machines on the network, access at
    http://turing.sickkids.ca :38000

**Comment by alipunjani** [comment]
Hey John,
The log file is at cryosparc/run/meteor.log 

meteor (the process) should never be running because that's only for development. The webapp process is what runs in deploy mode (but still logs to meteor.log) 

I'm not sure what it could be but might be related to the changes in the latest update. 
Thanks for reporting!

**Comment by johnrubinstein** [comment]
Hi Ali,
I don't see the logfile there. Is it wiped on restart? Or is it meteor.stdout that would have info (had a look but not sure what I'm seeing)

cryosparc@turing:~/cryosparc/run$ pwd
/home/cryosparc/cryosparc/run
cryosparc@turing:~/cryosparc/run$ ll
total 15300
drwxrwxr-x  7 cryosparc cryosparc     4096 Dec 13 18:02 ./
drwxrwxr-x 11 cryosparc cryosparc     4096 Dec  9 10:56 ../
drwxrwxr-x  2 cryosparc cryosparc     4096 Oct 24 11:28 bulk/
drwxrwxr-x  4 cryosparc cryosparc     4096 Nov 29 14:20 cache/
drwxrwxr-x  3 cryosparc cryosparc     4096 Dec 13 22:08 db/
-rw-r--r--  1 cryosparc cryosparc  3110113 Dec 13 22:07 meteor.stdout
-rw-r--r--  1 cryosparc cryosparc 12479292 Dec 13 21:36 mongod.stdout
drwxrwxr-x  3 cryosparc cryosparc     4096 Oct 24 11:04 sparcjobs/
-rw-rw-r--  1 cryosparc cryosparc    18668 Dec 13 18:02 supervisord.log
-rw-r--r--  1 cryosparc cryosparc        6 Dec 13 18:02 supervisord.pid
-rw-r--r--  1 cryosparc cryosparc     4356 Dec  9 10:56 update.log
drwxrwxr-x  2 cryosparc cryosparc     4096 Oct 24 11:04 upload_stage/

**Comment by alipunjani** [comment]
Sorry my mistake it's `meteor.stdout`

**Comment by alipunjani** [comment]
And also you are correct, the log file would have been erased on startup.. will fix that. For now I guess you have to wait until it crashes again and then look at/copy the logfile before starting again

**Comment by johnrubinstein** [comment]
Okay. I'll keep my eyes open and get it before restarting (and ask Hui and @zrip to do the same). Thx.

**Comment by alipunjani** [comment]
Aactually.. sorry my mistake again. The log file doesn't get cleared, it should all be there. Maybe you can just send the whole log file to me?

**Comment by johnrubinstein** [comment]
Here you are:
https://www.dropbox.com/s/slgavg6gaysqsja/meteor.stdout?dl=0

**Comment by alipunjani** [comment]
Thanks. Here's the error ( @suhaildawood any ideas? looks like a meteorhacks/kadira issue?)

```
{ job_id: 'vqS2QYPY2A2K7x9Es',
  createdAt: { '$gt': Fri Dec 02 2016 11:52:00 GMT-0500 (EST) } }
{ sort: { createdAt: 1 }, skip: 0, limit: 100 }
{ job_id: 'DYRBCDiDWtrND6iCq',
  createdAt: { '$gt': Sat Dec 10 2016 20:03:39 GMT-0500 (EST) } }
{ sort: { createdAt: 1 }, skip: 0, limit: 100 }
{ job_id: 'DYRBCDiDWtrND6iCq',
  createdAt: { '$gt': Sat Dec 10 2016 20:03:39 GMT-0500 (EST) } }
{ sort: { createdAt: 1 }, skip: 0, limit: 100 }
/home/cryosparc/cryosparc/cryosparc-webapp/bundle/programs/server/packages/meteorhacks_kadira.js:3545
  return originalRun.call(this, val);
                     ^

RangeError: Maximum call stack size exceeded
```

**Comment by alipunjani** [comment]
@johnrubinstein looks like the issues is in a javascript package that we were once using but is no longer needed, so we'll remove it in the next update. Thanks for catching this

**Comment by johnrubinstein** [comment]
Excellent. Thank you!

**Comment by mbrubake** [comment]
It would be great if we could resolve this soon.  I'm also seeing this here at OHSU.

**Comment by alipunjani** [comment]
This should be fixed now in Version 0.2.32
